test/system/reports_test.rb

```diff
@@ -688,4 +688,17 @@ def wait_for_watch_change
     assert_text 'おめでとう！'
     assert_text '100日目の日報を提出しました。'
   end
+
+  test 'should ignore unexpected class name to prevent XSS attack' do
+    visit_with_auth '/reports/new', 'komagata'
+    within('form[name=report]') do
+      fill_in('report[title]', with: 'test title')
+      fill_in('report[description]', with: ":::message \"></div><a href=\"javascript:alert('XSS');\">クリックしてね</a><div class=\"\nメッセージ\n:::")
+      fill_in('report[reported_on]', with: Time.current)
+
+      check '学習時間は無し', allow_label_click: true
+    end
+    click_button '提出'
+    assert_no_text 'クリックしてね'
```

Comment:
```suggestion
      fill_in('report[description]', with: ":::message \"></div><a href=\"javascript:alert('XSS');\">クリックしてね</a><div class=\"\nここにメッセージが入ります。\n:::")
      fill_in('report[reported_on]', with: Time.current)

      check '学習時間は無し', allow_label_click: true
    end
    click_button '提出'
    assert_text 'ここにメッセージが入ります。'
    assert_no_text 'クリックしてね'
```

「表示されないこと」だけでテストは終わらない方が良いです。

[システムスペックやフィーチャスペックで「〜が表示されないこと」だけを検証するのはちょっと危険、という話 - Qiita](https://qiita.com/jnchito/items/d5ba193af9b794dff77e)

あと、別件ですが、`:::message success`のように有効なclass名を指定した場合は、それが反映されることもテストできるとなお良しですね。

Reply:
チェックありがとうございます！異常系のテストを修正して、正常系の追加してみました。